conanfile.py

```diff
@@ -32,25 +32,20 @@ def set_version(self):
         self.version = re.search("set\(CMAKE_PROJECT_VERSION (.+)\)", content).group(1).strip()
 
     def configure(self):
-        # If we're building static then build the world as static, otherwise
-        # stuff will probably break.
-        # If you want your shared build to also build the world as shared, you
-        # can invoke conan with -o "&:shared=False" -o "*:shared=False",
-        # possibly with --build=missing or --build=cascade as desired,
-        # but this is probably not necessary.
-        if not self.options.shared:
-            self.options["*"].shared = False
+        # The shared-ness of grpc and protobuf needs to match our shared-ness or stuff will break
+        self.options["grpc"].shared = self.options.shared
+        self.options["protobuf"].shared = self.options.shared
```

Comment:
are grpc and protobuf the only libraries we care about? and if so how do we know that (just for my own understanding)

Reply:
Yeah so beforehand we were taking kind of a firehose approach of forcing the shared-ness of every single dependency which I think was not necessary. I'll elaborate on this in the doc comment but I think the thing that can potentially break is related to static/global data variable initializers being run by API which is calling into stuff from grpc, but maybe @acmorrow can confirm

Reply:
so are you saying there are potential issue when the API calls functions from gRPC, and those gRPC functions rely on static or global variables? and previously shared-ness was being forced on everything so this wasn't a problem?